jwst/cube_build/cube_cloud.py

```diff
@@ -44,224 +41,241 @@ def match_det2cube(self, input_model,
-                                    self.zdebug+1,ipt,xpix[ipt]+1,ypix[ipt]+1,
-                                    flux[ipt],file_slice_no,weight_distance,wave[ipt],zlam[iz],
-                                    d1*self.Cdelt1,d2*self.Cdelt2,d3*self.Cdelt3) +' \n')
-        iprint = iprint +1
-        if iprint == 10000:
-            log.debug('Mapping point and finding ROI for point cloud # %d %d' %(ipt,nn))
-            iprint = 0
+# If debug information is wanted find cube Location for later
+# comment out now for speed
+#________________________________________________________________________________
+#        if self.debug_pixel == 1:
+#            yy_cube = (indexr[0]/self.naxis1).astype(np.int)
+#            xx_cube = indexr[0] -yy_cube*self.naxis1
+#            cube_location = [ (xx,yy,zz) for zz in indexz[0] for xx,yy in zip(xx_cube,yy_cube)]
+
+#----------------------------------------------------------------------
+        # loop over all spaxels that overlap with ipt
+        # TODO: can this be done faster using some other method than looping
+        
+        for iz, zz in enumerate(icube_index):
+            spaxel[zz].flux = spaxel[zz].flux + weight_distance[iz] * flux[ipt]
+            spaxel[zz].flux_weight = spaxel[zz].flux_weight + weight_distance[iz]
+            spaxel[zz].iflux = spaxel[zz].iflux + 1
+#----------------------------------------------------------------------
+# debug informatons - comment out now for speed
+#            if( self.debug_pixel == 1 and self.xdebug == cube_location[iz][0] and
+#                self.ydebug == cube_location[iz][1] and
+#                self.zdebug ==cube_location[iz][2]):
+
+#                log.debug('For spaxel %d %d %d, %d detector x,y,flux %d %d %f %d %f '
+#                          %(self.xdebug+1,self.ydebug+1,
+#                            self.zdebug+1,ipt,xpix[ipt]+1,ypix[ipt]+1,
+#                            flux[ipt],file_slice_no,weight_distance[iz]))
+#                self.spaxel_debug.write(
+#                    'For spaxel %d %d %d %d, detector x,y,flux %d %d %f %d %f %f %f %f %f '
+#                    %(self.xdebug+1,self.ydebug+1,
+#                      self.zdebug+1,ipt,xpix[ipt]+1,ypix[ipt]+1,
+#                      flux[ipt],file_slice_no,weight_distance[iz],wave[ipt],
+#                      d1*self.Cdelt1,d2*self.Cdelt2,d3*self.Cdelt3) +' \n')
+
+#----------------------------------------------------------------------
+# comment out for speed
+#        iprint = iprint +1
+#        if iprint == 10000:
+#            log.debug('Mapping point and finding ROI for point cloud # %d %d' %(ipt,nn))
+#            iprint = 0
 #_______________________________________________________________________
 def FindWaveWeights(channel, subchannel):
     """
     Short Summary
     -------------
-    Get the wavelength normalization weights that we will 
+    Get the wavelength normalization weights that we will
```

Comment:
This function is currently not used anywhere. If it's here because it will be needed in the future, it may be faster to use a dict instead of a function.

Reply:
I will remove this. This was an old method that got replaced.